src/main/java/spireMapOverhaul/zones/manasurge/events/ManaCycloneEvent.java

```diff
@@ -0,0 +1,154 @@
+                        .cardSelectOption("Reached Inside",
+                                ()->{
+                                    CardGroup filteredCards = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+                                    AbstractDungeon.player.masterDeck.group.stream()
+                                            .filter(card -> !ManaSurgeZone.hasManaSurgeModifier(card) &&
+                                                    card.cost != -2 &&
+                                                    card.type != AbstractCard.CardType.CURSE &&
+                                                    card.type != AbstractCard.CardType.STATUS)
+                                            .forEach(filteredCards::addToTop);
+                                    return filteredCards;
+                                },
+                                "Select a card to enchant.",
+                                1, false, false, true, false,
+                                (cards)->{
+                                    for (AbstractCard c : cards) {
+                                        ManaSurgeZone.applyPermanentPositiveModifier(c);
+                                        CardCrawlGame.sound.play(ManaSurgeZone.ENCHANTBLIGHT_KEY);
+                                        AbstractDungeon.effectList.add(new ShowCardBrieflyEffect(c.makeStatEquivalentCopy()));
+                                    }
+                                    CardGroup filteredCards = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+                                    AbstractDungeon.player.masterDeck.group.stream()
+                                            .filter(card -> !ManaSurgeZone.hasManaSurgeModifier(card) &&
+                                                    card.cost != -2 &&
+                                                    card.type != AbstractCard.CardType.CURSE &&
+                                                    card.type != AbstractCard.CardType.STATUS)
+                                            .forEach(filteredCards::addToTop);
+
+                                    AbstractCard randomCard = filteredCards.getRandomCard(true);
+                                    if (randomCard != null) {
+                                        ManaSurgeZone.applyPermanentNegativeModifier(randomCard);
+                                        AbstractDungeon.effectList.add(new ShowCardBrieflyEffect(randomCard.makeStatEquivalentCopy()));
+                                        filteredCards.group.remove(randomCard);
+                                    }
+                                }
+                        )
+                )
+                .addOption(OPTIONS[1],(i) -> {
+                    int numberOfRelics = 6;
+                    int selectedRelicIndex = (int) (Math.random() * numberOfRelics);
+                    switch (selectedRelicIndex) {
+                        case 0:
```

Comment:
Let's not do this, there are a lot of issues here.
Instead, make a list of these relics IDs, filter all relics that the player has already (player.gasRelic(relicid), don't use string literals for the IDs use the static variable on the class like BottledTornado.ID

Then get a random string from that list and give the player that relic, if it is empty, give them some other relic (either custom or one that can stack so it doesn't matter if they have two copies of it)

Reply:
done I believe